tests/test_dense.py

```diff
@@ -177,3 +177,15 @@ def test_transform_fn(self):
                 del actual_data_array["linear_data_id"]
                 data_array_name = actual_data_array["name"]
                 self.assertEqual(actual_data_array, data_arrays[data_array_name])
+
+        """
+        Assures transform function creates gene data model correctly for files
+        with a number of genes that is a multiple of GENE_BATCH_SIZE (SCP-2669)
+        """
+        expression_matrix = DenseIngestor(
+            "../tests/data/dense_matrix_10_genes_15_cells.txt",
+            "5d276a50421aa9117c982845",
+            "5dd5ae25421aa910a723a337",
+        )
+        for actual_gene_models, actual_data_arrays in expression_matrix.transform():
+            self.assertEqual(DenseIngestor.GENE_BATCH_SIZE, len(actual_gene_models))
```

Comment:
I think this test should check that a total of 'x' cells (an amount equal to the amount of lines in the file -1) were processed as well that the batch size is **less than or equal** to the batch size. As we run into edge cases, these test files are updated to reflect the case. If someone adds additional genes or the amount of genes is not a multiple of 5 this test will fail.

Reply:
The strict equality is actually really important here.  If someone adds more genes to that file (so, e.g. it has 11 genes), it won't actually be testing what we want to test). As the comment says, this is for testing `files with a number of genes that is a multiple of GENE_BATCH_SIZE `  So a failure in this test would correctly tell someone they shouldn't have added genes to that file

Reply:
In that event we should still check for the first case scenario. And a comment should be added that the file should not be touched.

Reply:
I think what's important for test files is that the name includes the things that are important about the file -- e.g. 10_genes_15_cells  or error_headers.  If you're changing something that requires a name change to the file, that tells you you're changing something a test relies on.  So I'm not worried about this file being touched any more than I'd be worried about, e.g. someone updating the invalid_array_v2.1.2.txt file to not have any arrays.  Someone would have to both ignore the file name AND a failing test in order to make that change.